src/lib/blockchain_snark/blockchain_state.ml

```diff
@@ -81,9 +81,17 @@ module Make (Consensus_mechanism : Consensus.S) :
               |> Blockchain_state.snarked_ledger_hash )
               ( transition |> Snark_transition.blockchain_state
               |> Blockchain_state.snarked_ledger_hash )
+          and supply_increase_is_zero =
```

Comment:
You should use `equal`. `compare` costs roughly the number of bits while equal costs `1`.